api/src/test/java/com/ford/labs/retroquest/columntitle/ColumnTitleServiceTest.java

```diff
@@ -0,0 +1,70 @@
+package com.ford.labs.retroquest.columntitle;
+
+import com.ford.labs.retroquest.exception.ColumnTitleNotFoundException;
+import io.micrometer.core.instrument.MeterRegistry;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@SpringBootTest
+class ColumnTitleServiceTest {
+
+    @Autowired
+    ColumnTitleRepository columnTitleRepository;
+
+    @Autowired
+    MeterRegistry meterRegistry;
+
+    @Autowired
+    ColumnTitleService columnTitleService;
+
+    @BeforeEach
+    void beforeEach() {
+        columnTitleRepository.deleteAll();
+    }
+
+    @AfterEach
+    void afterEach() {
+        columnTitleRepository.deleteAll();
+    }
+
+    private final ColumnTitle.ColumnTitleBuilder columnTitleBuilder = ColumnTitle.builder();
+
+    @Test
+    void given_teamid_get_column_titles() {
+        String teamId = "some team id";
+        String teamId2 = "some other team id";
+        var columnTitle1 = columnTitleRepository.save(columnTitleBuilder.topic("happy").title("Some Title").teamId(teamId).build());
+        var columnTitle2 = columnTitleRepository.save(columnTitleBuilder.topic("meh").title("Some Other Title").teamId(teamId).build());
+        var columnTitle3 = columnTitleRepository.save(columnTitleBuilder.topic("happy").title("Some Title").teamId(teamId2).build());
+
+        assertThat(columnTitleService.getColumnTitlesByTeamId(teamId)).containsExactlyInAnyOrder(columnTitle1, columnTitle2);
+        assertThat(columnTitleService.getColumnTitlesByTeamId(teamId2)).containsExactlyInAnyOrder(columnTitle3);
+    }
+
+    @Test
+    void given_column_id_and_column_title_rename_column_in_db_and_return_new_column_title() {
+        String teamId = "some team ID";
+        var columnTitle = columnTitleRepository.save(columnTitleBuilder.topic("happy").title("Some Title").teamId(teamId).build());
+
+        String newColumnName = "Some new Title";
+        var savedColumnTitle = columnTitleService.editColumnTitleName(columnTitle.getId(), newColumnName, teamId );
+
+        assertThat(meterRegistry.get("retroquest.columns.changed.count").counter().count()).isEqualTo(1);
+        assertThat(savedColumnTitle.getTitle()).isEqualTo(newColumnName);
+        assertThat(savedColumnTitle.getId()).isEqualTo(columnTitle.getId());
+    }
+
+    @Test
+    void throws_column_title_not_found_exception_when_column_title_not_in_db() {
```

Comment:
Uhh, I guess you could? but it returns a list, so it would just send an empty list back. It's kind of a weird edge case, because column titles are generated when you make a team; so it would presumably be caused by trying to look up the titles for a team that doesn't exist.

Reply:
I'll make a card for this then. There are other spots in the app where this is a problem too.